File: mobile/src/main/java/net/mm2d/android/view/TransitionListenerAdapter.java
/*
 * Copyright (c) 2017 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.android.view;

import android.os.Build;
import android.transition.Transition;

import androidx.annotation.RequiresApi;

/**
 * Transition.TransitionListenerの空実装。
 *
 * @author <a href="mailto:ryo@mm2d.net">大前良介 (OHMAE Ryosuke)</a>
 */
@RequiresApi(api = Build.VERSION_CODES.KITKAT)
public class TransitionListenerAdapter implements Transition.TransitionListener {
    @Override
    public void onTransitionStart(Transition transition) {
    }

    @Override
    public void onTransitionEnd(Transition transition) {
    }

    @Override
    public void onTransitionCancel(Transition transition) {
    }

    @Override
    public void onTransitionPause(Transition transition) {
    }

    @Override
    public void onTransitionResume(Transition transition) {
    }
}

File: mobile/src/main/kotlin/net/mm2d/android/view/TransitionListenerAdapter.kt
/*
 * Copyright (c) 2017 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.android.view

import android.transition.Transition

/**
 * Transition.TransitionListenerの空実装。
 *
 * @author [大前良介 (OHMAE Ryosuke)](mailto:ryo@mm2d.net)
 */
open class TransitionListenerAdapter : Transition.TransitionListener {
    override fun onTransitionStart(
        transition: Transition,
    ) = Unit

    override fun onTransitionEnd(
        transition: Transition,
    ) = Unit

    override fun onTransitionCancel(
        transition: Transition,
    ) = Unit

    override fun onTransitionPause(
        transition: Transition,
    ) = Unit

    override fun onTransitionResume(
        transition: Transition,
    ) = Unit
}
